### PR TITLE
refactor(db/schema): store documentation of fields alongside code

### PR DESCRIPTION
**Pull request by jschmid1**
### Summary

Supersedes #10706 
Supersedes https://github.com/Kong/kong/pull/10150

This PR is designed to streamline our documentation process for all configurable fields within Kong, by positioning the documentation closer to their respective definitions. This strategic placement facilitates automatic generation of updated documentation, eliminating the need for multiple location updates.

At this point, I didn't mark the description field `required = true`, in order to prevent potential conflicts when transitioning to EE. Instead, we should approach this as an iterative process, only making description fields mandatory once we are confident of their completeness.

Co-authored-by: Angel Guarisma angel.guarisma@konghq.com

### Checklist

~- [ ] The Pull Request has tests~
~- [ ] There's an entry in the CHANGELOG~
~- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE~

### Full changelog

* adding `description` fields to all configurable fields.

### Issue reference

https://konghq.atlassian.net/browse/KAG-475

### CODE REVIEW

File: kong/db/schema/metaschema.lua [hunk 1]
```diff
@@ -72,6 +72,7 @@ local field_schema = {
   { type = { type = "string", one_of = keys(Schema.valid_types), required = true }, },
   { required = { type = "boolean" }, },
   { reference = { type = "string" }, },
+  { description = { type = "string" }, },
```
Comment (hbagdi) [inline kong/db/schema/metaschema.lua:75]: Let's add a minimum and a maximum length limit to these.

Comment (jschmid1) [reply kong/db/schema/metaschema.lua]: Added a minimum length of `10` and a maximum length of `500` characters. This tolerates the _existing_ descriptions we have in our docs (with one needed change). The longest description we have is 447 characters long.

As a reference here is an example text with 500 characters:

```
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla efficitur urna neque, eu vehicula libero tristique ac. Vivamus lacinia turpis vel mi feugiat, eu pharetra dui facilisis. Sed in sapien id odio bibendum fermentum nec ut justo. Aenean imperdiet lacinia nisi, nec tempus odio euismod ut. Pellentesque at magna ac ligula commodo aliquet eget a arcu. Proin vitae dolor vel mi vulputate condimentum. Suspendisse bibendum purus velit, ut condimentum odio finibus nec. Fusce scelerisque, quam ut sollicitudin fringilla, sapien mauris tristique odio, vel dictum lectus libero id ante. Integer ullamcorper malesuada sapien, non gravida sapien lobortis vel. Etiam euismod, lorem nec cursus hendrerit, lectus nulla pharetra augue, id pulvinar dui lacus eu turpis. In hac habitasse platea dictumst. Ut sit amet tortor nunc.
```

And an example text with 10 characters.

```
Lorem ipsum d
```

Comment (hbagdi) [reply kong/db/schema/metaschema.lua]: Looks good.

**Comment by hbagdi** [comment]
@jschmid1 Please rebase.

**Comment by hbagdi** [comment]
@chronolaw Can you please fix the failing test?

**Comment by Guaris** [comment]
@hbagdi @kikito @jschmid1 Can we merge this?

File: kong/plugins/response-ratelimiting/schema.lua [hunk 2]
```diff
@@ -36,16 +36,19 @@ end
...
     one_of = {
       "local",
       "redis",
     },
   }
-
 else
   policy = {
+    description =
+    "The rate-limiting policies to use for retrieving and incrementing the\nlimits.",
```
Comment (kikito) [inline kong/plugins/response-ratelimiting/schema.lua:51]:
```suggestion
    "The rate-limiting policies to use for retrieving and incrementing the limits.",
```

File: kong/plugins/statsd/schema.lua [hunk 3]
```diff
@@ -181,8 +188,7 @@ return {
                 },
               },
           }, },
-          { allow_status_codes = {
-            type = "array",
+          { allow_status_codes = { description = "List of status code ranges that are allowed to be logged in metrics.", type = "array",
```
Comment (kikito) [inline kong/plugins/statsd/schema.lua:191]: The style on the field descriptions on this plugin is a bit inconsistent:
* Using "The xxx that does yyy" formula: "The IP address ..."
* Just saying "Does yyy": "Determines what sort of event... "
* Just saying "xxx": "List of status code ..."

It seems like the first style ("The xxx that does yyy") is the most usual one. I propose using that style in all the fields.

File: kong/tools/queue_schema.lua [hunk 4]
```diff
@@ -8,46 +8,46 @@ return Schema.define {
       type = "integer",
       default = 1,
       between = { 1, 1000000 },
-      -- description = "maximum number of entries that can be processed at a time"
+      description = "maximum number of entries that can be processed at a time"
```
Comment (kikito) [inline kong/tools/queue_schema.lua:11]: I think these descriptions should start in Uppercase and end in a dot.

```suggestion
      description = "Maximum number of entries that can be processed at a time."
```

**Review by kikito — CHANGES_REQUESTED** [review]
I think this PR is mostly fine, my only real complain is with the "FILL ME WITH SOMETHING MEANINGFUL" descriptions, we should not merge this with those in, unless there's a valid reason

File: kong/db/schema/entities/clustering_data_planes.lua [hunk 5]
```diff
@@ -29,6 +29,7 @@ return {
                       required = true,
                       one_of = SYNC_STATUS_CHOICES,
                       default = "unknown",
+                      description = "The status of the clustering data planes sync",
```
Comment (Guaris) [inline kong/db/schema/entities/clustering_data_planes.lua:32]:
```suggestion
                      description = "The status of the clustering data planes sync.",
```

File: kong/db/schema/entities/consumers.lua [hunk 6]
```diff
@@ -1,18 +1,32 @@
...
+        description =
+        "The unique username of the Consumer. You must send either this field or custom_id with the request.",
+        type = "string",
+        unique = true
+      },
+    },
+    {
+      custom_id =
+      {
+        description = "Field for storing an existing unique ID for the Consumer",
```
Comment (Guaris) [inline kong/db/schema/entities/consumers.lua:24]:
```suggestion
        description = "Stores the existing unique ID of the consumer.",
```

File: kong/db/schema/entities/parameters.lua [hunk 7]
```diff
@@ -9,7 +9,7 @@ return {
 
   fields = {
     { created_at     = typedefs.auto_timestamp_s },
-    { key            = { type = "string", required = true, unique = true, }, },
-    { value          = { type = "string", required = true, }, },
+    { key            = { description = "They key value of a parameter", type = "string", required = true, unique = true, }, },
```
Comment (Guaris) [inline kong/db/schema/entities/parameters.lua:12]:
```suggestion
    { key            = { description = "They key value of a parameter.", type = "string", required = true, unique = true, }, },
```

File: kong/db/schema/entities/parameters.lua [hunk 8]
```diff
@@ -9,7 +9,7 @@ return {
...
+    { value          = { description = "The value attached to the key", type = "string", required = true, }, },
```
Comment (Guaris) [inline kong/db/schema/entities/parameters.lua:13]:
```suggestion
    { value          = { description = "The value attached to the key.", type = "string", required = true, }, },
```

File: kong/db/schema/entities/plugins.lua [hunk 9]
```diff
@@ -15,16 +15,16 @@ return {
...
     { instance_name = typedefs.utf8_name },
     { created_at = typedefs.auto_timestamp_s },
     { updated_at = typedefs.auto_timestamp_s },
-    { route = { type = "foreign", reference = "routes", default = null, on_delete = "cascade", }, },
-    { service = { type = "foreign", reference = "services", default = null, on_delete = "cascade", }, },
-    { consumer = { type = "foreign", reference = "consumers", default = null, on_delete = "cascade", }, },
-    { config = { type = "record", abstract = true, }, },
+    { route = { description = "If set, the plugin will only activate when receiving requests via the specified route.", type = "foreign", reference = "routes", default = null, on_delete = "cascade", }, },
+    { service = { description = "If set, the plugin will only activate when receiving requests via one of the routes belonging to the specified service. ", type = "foreign", reference = "services", default = null, on_delete = "cascade", }, },
+    { consumer = { description = "If set, the plugin will activate only for requests where the specified has been authenticated. ", type = "foreign", reference = "consumers", default = null, on_delete = "cascade", }, },
```
Comment (Guaris) [inline kong/db/schema/entities/plugins.lua:24]:
```suggestion
    { consumer = { description = "If set, the plugin will activate only for requests where the specified has been authenticated.", type = "foreign", reference = "consumers", default = null, on_delete = "cascade", }, },
```

File: kong/db/schema/entities/plugins.lua [hunk 10]
```diff
@@ -15,16 +15,16 @@ return {
...
+    { config = { description = "The configuration properties for the Plugin", type = "record", abstract = true, }, },
```
Comment (Guaris) [inline kong/db/schema/entities/plugins.lua:25]:
```suggestion
    { config = { description = "The configuration properties for the Plugin.", type = "record", abstract = true, }, },
```

File: kong/db/schema/entities/routes.lua [hunk 11]
```diff
@@ -43,17 +44,18 @@ if kong_router_flavor == "expressions" then
...
+                                       description = "The status code Kong responds with when all properties of a Route match except the protocol",
                                        one_of = { 426, 301, 302, 307, 308 },
                                        default = 426, required = true,
                                      }, },
-      { strip_path     = { type = "boolean", required = true, default = true }, },
-      { preserve_host  = { type = "boolean", required = true, default = false }, },
-      { request_buffering  = { type = "boolean", required = true, default = true }, },
-      { response_buffering  = { type = "boolean", required = true, default = true }, },
+      { strip_path     = { description = "When matching a Route via one of the paths, strip the matching prefix from the upstream request URL.", type = "boolean", required = true, default = true }, },
+      { preserve_host  = { description = "When matching a Route via one of the hosts domain names, use the request Host header in the upstream request headers. ", type = "boolean", required = true, default = false }, },
```
Comment (Guaris) [inline kong/db/schema/entities/routes.lua:52]:
```suggestion
      { preserve_host  = { description = "When matching a Route via one of the hosts domain names, use the request Host header in the upstream request headers.", type = "boolean", required = true, default = false }, },
```

File: kong/db/schema/entities/routes.lua [hunk 12]
```diff
@@ -43,17 +44,18 @@ if kong_router_flavor == "expressions" then
...
+      { request_buffering  = { description = "Whether to enable request body buffering or not. With HTTP 1.1,", type = "boolean", required = true, default = true }, },
```
Comment (Guaris) [inline kong/db/schema/entities/routes.lua:53]:
```suggestion
      { request_buffering  = { description = "Whether to enable request body buffering or not. With HTTP 1.1.", type = "boolean", required = true, default = true }, },
```

File: kong/db/schema/entities/routes.lua [hunk 13]
```diff
@@ -43,17 +44,18 @@ if kong_router_flavor == "expressions" then
...
+      { response_buffering  = { description = "Whether to enable response body buffering or not.", type = "boolean", required = true, default = true }, },
       { tags             = typedefs.tags },
-      { service = { type = "foreign", reference = "services" }, },
-      { expression = { type = "string", required = true }, },
-      { priority = { type = "integer", required = true, default = 0 }, },
+      { service = { description = "The Service this Route is associated to. This is where the Route proxies traffic to.", type = "foreign", reference = "services" }, },
+      { expression = { description = " The router expression", type = "string", required = true }, },
```
Comment (Guaris) [inline kong/db/schema/entities/routes.lua:57]:
```suggestion
      { expression = { description = " The router expression.", type = "string", required = true }, },
```

File: kong/db/schema/entities/routes.lua [hunk 14]
```diff
@@ -111,21 +114,24 @@ else
...
-      { request_buffering  = { type = "boolean", required = true, default = true }, },
-      { response_buffering  = { type = "boolean", required = true, default = true }, },
+      { regex_priority = { description = "A number used to choose which route resolves a given request when several routes match it using regexes simultaneously.", type = "integer", default = 0 }, },
+      { strip_path     = { description = "When matching a Route via one of the paths, strip the matching prefix from the upstream request URL.", type = "boolean", required = true, default = true }, },
+      { path_handling  = { description = "Controls how the Service path, Route path and requested path are combined when sending a request to the upstream.", type = "string", default = "v0", one_of = { "v0", "v1" }, }, },
+      { preserve_host  = { description = "When matching a Route via one of the hosts domain names, use the request Host header in the upstream request headers.", type = "boolean", required = true, default = false }, },
+      { request_buffering  = { description = "Whether to enable request body buffering or not. With HTTP 1.1", type = "boolean", required = true, default = true }, },
```
Comment (Guaris) [inline kong/db/schema/entities/routes.lua:125]:
```suggestion
      { request_buffering  = { description = "Whether to enable request body buffering or not. With HTTP 1.1.", type = "boolean", required = true, default = true }, },
```

File: kong/db/schema/entities/routes.lua [hunk 15]
```diff
@@ -111,21 +114,24 @@ else
...
+      { response_buffering  = { description = "Whether to enable response body buffering or not. ", type = "boolean", required = true, default = true }, },
```
Comment (Guaris) [inline kong/db/schema/entities/routes.lua:126]:
```suggestion
      { response_buffering  = { description = "Whether to enable response body buffering or not.", type = "boolean", required = true, default = true }, },
```

File: kong/db/schema/entities/services.lua [hunk 16]
```diff
@@ -39,11 +40,11 @@ return {
     { write_timeout      = nonzero_timeout { default = 60000 }, },
     { read_timeout       = nonzero_timeout { default = 60000 }, },
     { tags               = typedefs.tags },
-    { client_certificate = { type = "foreign", reference = "certificates" }, },
-    { tls_verify         = { type = "boolean", }, },
-    { tls_verify_depth   = { type = "integer", default = null, between = { 0, 64 }, }, },
-    { ca_certificates    = { type = "array", elements = { type = "string", uuid = true, }, }, },
-    { enabled            = { type = "boolean", required = true, default = true, }, },
+    { client_certificate = { description = "Certificate to be used as client certificate while TLS handshaking to the upstream server.", type = "foreign", reference = "certificates" }, },
+    { tls_verify         = { description = "Whether to enable verification of upstream server TLS certificate. ", type = "boolean", }, },
```
Comment (Guaris) [inline kong/db/schema/entities/services.lua:44]:
```suggestion
    { tls_verify         = { description = "Whether to enable verification of upstream server TLS certificate.", type = "boolean", }, },
```

File: kong/db/schema/entities/services.lua [hunk 17]
```diff
@@ -39,11 +40,11 @@ return {
...
+    { tls_verify_depth   = { description = "Maximum depth of chain while verifying Upstream server's TLS certificate", type = "integer", default = null, between = { 0, 64 }, }, },
```
Comment (Guaris) [inline kong/db/schema/entities/services.lua:45]:
```suggestion
    { tls_verify_depth   = { description = "Maximum depth of chain while verifying Upstream server's TLS certificate.", type = "integer", default = null, between = { 0, 64 }, }, },
```

File: kong/db/schema/entities/snis.lua [hunk 18]
```diff
@@ -14,7 +14,7 @@ return {
     { created_at   = typedefs.auto_timestamp_s },
     { updated_at   = typedefs.auto_timestamp_s },
     { tags         = typedefs.tags },
-    { certificate  = { type = "foreign", reference = "certificates", required = true }, },
+    { certificate  = { description = "The id (a UUID) of the certificate with which to associate the SNI hostname. ", type = "foreign", reference = "certificates", required = true }, },
```
Comment (Guaris) [inline kong/db/schema/entities/snis.lua:17]:
```suggestion
    { certificate  = { description = "The id (a UUID) of the certificate with which to associate the SNI hostname.", type = "foreign", reference = "certificates", required = true }, },
```

File: kong/db/schema/entities/tags.lua [hunk 19]
```diff
@@ -9,7 +9,7 @@ return {
 
   fields = {
     { tag          = typedefs.tag, },
-    { entity_name  = { type = "string", required = true }, },
+    { entity_name  = { description = "The name of the Kong Gateway entity being tagged", type = "string", required = true }, },
```
Comment (Guaris) [inline kong/db/schema/entities/tags.lua:12]:
```suggestion
    { entity_name  = { description = "The name of the Kong Gateway entity being tagged.", type = "string", required = true }, },
```

File: kong/db/schema/entities/targets.lua [hunk 20]
```diff
@@ -27,9 +27,9 @@ return {
     { id = typedefs.uuid },
     { created_at = typedefs.auto_timestamp_ms },
     { updated_at = typedefs.auto_timestamp_ms },
-    { upstream   = { type = "foreign", reference = "upstreams", required = true, on_delete = "cascade" }, },
-    { target     = { type = "string", required = true, custom_validator = validate_target, }, },
-    { weight     = { type = "integer", default = 100, between = { 0, 65535 }, }, },
+    { upstream   = { description = "The unique identifier or the name of the upstream for which to update the target.", type = "foreign", reference = "upstreams", required = true, on_delete = "cascade" }, },
+    { target     = { description = "The target address (ip or hostname) and port. ", type = "string", required = true, custom_validator = validate_target, }, },
```
Comment (Guaris) [inline kong/db/schema/entities/targets.lua:31]:
```suggestion
    { target     = { description = "The target address (ip or hostname) and port.", type = "string", required = true, custom_validator = validate_target, }, },
```

File: kong/db/schema/entities/upstreams.lua [hunk 21]
```diff
@@ -179,31 +179,31 @@ local r =  {
...
+    { algorithm = { description = "Which load balancing algorithm to use.", type = "string",
         default = "round-robin",
         one_of = { "consistent-hashing", "least-connections", "round-robin", "latency" },
     }, },
     { hash_on = hash_on },
     { hash_fallback = hash_on },
     { hash_on_header = typedefs.header_name, },
     { hash_fallback_header = typedefs.header_name, },
-    { hash_on_cookie = { type = "string",  custom_validator = utils.validate_cookie_name }, },
+    { hash_on_cookie = { description = "The cookie name to take the value from as hash input. ", type = "string",  custom_validator = utils.validate_cookie_name }, },
```
Comment (Guaris) [inline kong/db/schema/entities/upstreams.lua:192]:
```suggestion
    { hash_on_cookie = { description = "The cookie name to take the value from as hash input.", type = "string",  custom_validator = utils.validate_cookie_name }, },
```

File: kong/db/schema/entities/upstreams.lua [hunk 22]
```diff
@@ -179,31 +179,31 @@ local r =  {
...
     { hash_on_cookie_path = typedefs.path{ default = "/", }, },
     { hash_on_query_arg = simple_param },
     { hash_fallback_query_arg = simple_param },
     { hash_on_uri_capture = simple_param },
     { hash_fallback_uri_capture = simple_param },
-    { slots = { type = "integer", default = 10000, between = { 10, 2^16 }, }, },
-    { healthchecks = { type = "record",
+    { slots = { description = "The number of slots in the load balancer algorithm. ", type = "integer", default = 10000, between = { 10, 2^16 }, }, },
```
Comment (Guaris) [inline kong/db/schema/entities/upstreams.lua:198]:
```suggestion
    { slots = { description = "The number of slots in the load balancer algorithm.", type = "integer", default = 10000, between = { 10, 2^16 }, }, },
```

File: kong/db/schema/entities/upstreams.lua [hunk 23]
```diff
@@ -179,31 +179,31 @@ local r =  {
...
+    { healthchecks = { description = "The array of healthchecks", type = "record",
```
Comment (Guaris) [inline kong/db/schema/entities/upstreams.lua:199]:
```suggestion
    { healthchecks = { description = "The array of healthchecks.", type = "record",
```

File: kong/db/schema/entities/upstreams.lua [hunk 24]
```diff
@@ -179,31 +179,31 @@ local r =  {
...
         default = healthchecks_defaults,
         fields = healthchecks_fields,
     }, },
     { tags = typedefs.tags },
     { host_header = typedefs.host_with_optional_port },
-    { client_certificate = { type = "foreign", reference = "certificates" }, },
-    { use_srv_name =  { type = "boolean", default = false, }, },
+    { client_certificate = { description = "If set, the certificate to be used as client certificate while TLS handshaking to the upstream server.", type = "foreign", reference = "certificates" }, },
+    { use_srv_name =  { description = "If set, the balancer will use SRV hostname", type = "boolean", default = false, }, },
```
Comment (Guaris) [inline kong/db/schema/entities/upstreams.lua:206]:
```suggestion
    { use_srv_name =  { description = "If set, the balancer will use SRV hostname.", type = "boolean", default = false, }, },
```

File: kong/db/schema/entities/vaults.lua [hunk 25]
```diff
@@ -55,11 +55,11 @@ return {
     { id = typedefs.uuid },
     -- note: prefix must be valid in a host part of vault reference uri:
     -- {vault://<vault-prefix>/<secret-id>[/<secret-key]}
-    { prefix = { type = "string", required = true, unique = true, unique_across_ws = true,
+    { prefix = { description = "The unique prefix (or identifier) for this Vault configuration.", type = "string", required = true, unique = true, unique_across_ws = true,
                  match = [[^[a-z][a-z%d-]-[a-z%d]+$]], not_one_of = VAULTS }},
-    { name = { type = "string", required = true }},
-    { description = { type = "string" }},
-    { config = { type = "record", abstract = true }},
+    { name = { description = "The name of the Vault that's going to be added. ", type = "string", required = true }},
```
Comment (Guaris) [inline kong/db/schema/entities/vaults.lua:60]:
```suggestion
    { name = { description = "The name of the Vault that's going to be added.", type = "string", required = true }},
```

File: kong/db/schema/typedefs.lua [hunk 26]
```diff
@@ -305,110 +321,130 @@ typedefs.path = Schema.define {
...
 typedefs.sni = Schema.define {
   type = "string",
   custom_validator = validate_sni,
+  description = "A string representing an SNI (server name indication) value for TLS."
+}
+
+typedefs.redis_server_name = Schema.define {
+  type = "string",
+  custom_validator = validate_sni,
+  description = "When using the `redis` policy with `redis_ssl` set to `true`, this property specifies the server name for the TLS extension Server Name Indication (SNI) "
```
Comment (Guaris) [inline kong/db/schema/typedefs.lua:420]:
```suggestion
  description = "When using the `redis` policy with `redis_ssl` set to `true`, this property specifies the server name for the TLS extension Server Name Indication (SNI)."
```

File: kong/plugins/acme/schema.lua [hunk 27]
```diff
@@ -34,42 +37,50 @@ local KONG_STORAGE_SCHEMA = {
...
 }
 
 local CONSUL_STORAGE_SCHEMA = {
-  { https = { type = "boolean", default = false, }, },
-  { host = typedefs.host, },
-  { port = typedefs.port, },
-  { kv_path = { type = "string", }, },
-  { timeout = { type = "number", }, },
-  { token = { type = "string", referenceable = true, }, },
+  { https = { type = "boolean", default = false, description = "FILL ME WITH SOMETHING MEANINGFUL"}, },
```
Comment (Guaris) [inline kong/plugins/acme/schema.lua:58]:
```suggestion
  { https = { type = "boolean", default = false, description = "Boolean representation of https."}, },
```

File: kong/plugins/acme/schema.lua [hunk 28]
```diff
@@ -34,42 +37,50 @@ local KONG_STORAGE_SCHEMA = {
...
+  { host = typedefs.host},
+  { port = typedefs.port},
+  { kv_path = { type = "string", description = "FILL ME WITH SOMETHING MEANINGFUL"}, },
```
Comment (Guaris) [inline kong/plugins/acme/schema.lua:61]:
```suggestion
  { kv_path = { type = "string", description = "KV prefix path."}, },
```

File: kong/plugins/acme/schema.lua [hunk 29]
```diff
@@ -34,42 +37,50 @@ local KONG_STORAGE_SCHEMA = {
...
+  { timeout = { type = "number", description = "FILL ME WITH SOMETHING MEANINGFUL"}, },
```
Comment (Guaris) [inline kong/plugins/acme/schema.lua:62]:
```suggestion
  { timeout = { type = "number", description = "Timeout in milliseconds."}, },
```

File: kong/plugins/acme/schema.lua [hunk 30]
```diff
@@ -34,42 +37,50 @@ local KONG_STORAGE_SCHEMA = {
...
+  { token = { type = "string", referenceable = true, description = "FILL ME WITH SOMETHING MEANINGFUL"}, },
```
Comment (Guaris) [inline kong/plugins/acme/schema.lua:63]:
```suggestion
  { token = { type = "string", referenceable = true, description = "Consul ACL token."}, },
```

File: kong/plugins/acme/schema.lua [hunk 31]
```diff
@@ -34,42 +37,50 @@ local KONG_STORAGE_SCHEMA = {
...
 }
 
 local VAULT_STORAGE_SCHEMA = {
-  { https = { type = "boolean", default = false, }, },
+  { https = { type = "boolean", default = false, description = "FILL ME WITH SOMETHING MEANINGFUL" }, },
```
Comment (Guaris) [inline kong/plugins/acme/schema.lua:67]:
```suggestion
  { https = { type = "boolean", default = false, description = "Boolean representation of https." }, },
```

File: kong/plugins/acme/schema.lua [hunk 32]
```diff
@@ -34,42 +37,50 @@ local KONG_STORAGE_SCHEMA = {
...
-  { kv_path = { type = "string", }, },
-  { timeout = { type = "number", }, },
-  { token = { type = "string", referenceable = true, }, },
-  { tls_verify = { type = "boolean", default = true, }, },
-  { tls_server_name = { type = "string" }, },
-  { auth_method = { type = "string", default = "token", one_of = { "token", "kubernetes" } } },
-  { auth_path =  { type = "string" }, },
-  { auth_role =  { type = "string" }, },
-  { jwt_path =  { type = "string" }, },
+  { kv_path = { type = "string", description = "FILL ME WITH SOMETHING MEANINGFUL" }, },
```
Comment (Guaris) [inline kong/plugins/acme/schema.lua:70]:
```suggestion
  { kv_path = { type = "string", description = "KV prefix path." }, },
```

File: kong/plugins/acme/schema.lua [hunk 33]
```diff
@@ -34,42 +37,50 @@ local KONG_STORAGE_SCHEMA = {
...
+  { timeout = { type = "number", description = "FILL ME WITH SOMETHING MEANINGFUL"}, },
```
Comment (Guaris) [inline kong/plugins/acme/schema.lua:71]:
```suggestion
  { timeout = { type = "number", description = "Timeout in milliseconds."}, },
```

File: kong/plugins/acme/schema.lua [hunk 34]
```diff
@@ -34,42 +37,50 @@ local KONG_STORAGE_SCHEMA = {
...
+  { token = { type = "string", referenceable = true, description = "FILL ME WITH SOMETHING MEANINGFUL" }, },
```
Comment (Guaris) [inline kong/plugins/acme/schema.lua:72]:
```suggestion
  { token = { type = "string", referenceable = true, description = "Consul ACL token." }, },
```

File: kong/plugins/acme/schema.lua [hunk 35]
```diff
@@ -34,42 +37,50 @@ local KONG_STORAGE_SCHEMA = {
...
+  { tls_verify = { type = "boolean", default = true, description = "FILL ME WITH SOMETHING MEANINGFUL" }, },
```
Comment (Guaris) [inline kong/plugins/acme/schema.lua:73]:
```suggestion
  { tls_verify = { type = "boolean", default = true, description = "Turn on TLS verification." }, },
```

File: kong/plugins/acme/schema.lua [hunk 36]
```diff
@@ -34,42 +37,50 @@ local KONG_STORAGE_SCHEMA = {
...
+  { tls_server_name = { type = "string", description = "FILL ME WITH SOMETHING MEANINGFUL"  }, },
```
Comment (Guaris) [inline kong/plugins/acme/schema.lua:74]:
```suggestion
  { tls_server_name = { type = "string", description = "SNI used in request, default to host if omitted."  }, },
```

File: kong/plugins/acme/schema.lua [hunk 37]
```diff
@@ -34,42 +37,50 @@ local KONG_STORAGE_SCHEMA = {
...
+  { auth_method = { type = "string", default = "token", one_of = { "token", "kubernetes" }, description = "FILL ME WITH SOMETHING MEANINGFUL" } },
```
Comment (Guaris) [inline kong/plugins/acme/schema.lua:75]:
```suggestion
  { auth_method = { type = "string", default = "token", one_of = { "token", "kubernetes" }, description = "Auth Method, default to token, can be "token" or "kubernetes." } },
```

File: kong/plugins/acme/schema.lua [hunk 38]
```diff
@@ -34,42 +37,50 @@ local KONG_STORAGE_SCHEMA = {
...
+  { auth_path = { type = "string", description = "FILL ME WITH SOMETHING MEANINGFUL" }, },
```
Comment (Guaris) [inline kong/plugins/acme/schema.lua:76]:
```suggestion
  { auth_path = { type = "string", description = "Vault's authentication path to use." }, },
```

File: kong/plugins/acme/schema.lua [hunk 39]
```diff
@@ -34,42 +37,50 @@ local KONG_STORAGE_SCHEMA = {
...
+  { auth_role = { type = "string", description = "FILL ME WITH SOMETHING MEANINGFUL" }, },
```
Comment (Guaris) [inline kong/plugins/acme/schema.lua:77]:
```suggestion
  { auth_role = { type = "string", description = "The role to try and assign." }, },
```

File: kong/plugins/acme/schema.lua [hunk 40]
```diff
@@ -34,42 +37,50 @@ local KONG_STORAGE_SCHEMA = {
...
+  { jwt_path = { type = "string", description = "FILL ME WITH SOMETHING MEANINGFUL" }, },
```
Comment (Guaris) [inline kong/plugins/acme/schema.lua:78]:
```suggestion
  { jwt_path = { type = "string", description = "The path to the JWT." }, },
```

File: kong/plugins/acme/schema.lua [hunk 41]
```diff
@@ -34,42 +37,50 @@ local KONG_STORAGE_SCHEMA = {
...
 }
 
 local ACCOUNT_KEY_SCHEMA = {
-  { key_id = { type = "string", required = true }},
-  { key_set = { type = "string" }}
+  { key_id = { type = "string", required = true, description = "FILL ME WITH SOMETHING MEANINGFUL" } },
```
Comment (Guaris) [inline kong/plugins/acme/schema.lua:82]:
```suggestion
  { key_id = { type = "string", required = true, description = "The Key ID." } },
```

File: kong/plugins/acme/schema.lua [hunk 42]
```diff
@@ -34,42 +37,50 @@ local KONG_STORAGE_SCHEMA = {
...
+  { key_set = { type = "string", description = "FILL ME WITH SOMETHING MEANINGFUL" } }
```
Comment (Guaris) [inline kong/plugins/acme/schema.lua:83]:
```suggestion
  { key_set = { type = "string", description = "The ID of the key set to associate the Key ID with." } }
```

File: kong/plugins/acme/schema.lua (hunk 27 above)
[the remark this answers is not in the document]
Comment (Guaris) [reply kong/plugins/acme/schema.lua:58]: I have just added to fix this.

File: kong/plugins/statsd/schema.lua (hunk 3 above)
Comment (jschmid1) [reply kong/plugins/statsd/schema.lua:191]: @Guaris Your call.

File: kong/tools/queue_schema.lua [hunk 43]
```diff
@@ -8,46 +8,46 @@ return Schema.define {
...
       between = { 1, 1000000 },
-      -- description = "maximum number of entries that can be processed at a time"
+      description = "Maximum number of entries that can be processed at a time."
     } },
     { max_coalescing_delay = {
       type = "number",
       default = 1,
       between = { 0, 3600 },
-      -- description = "maximum number of (fractional) seconds to elapse after the first entry was queued before the queue starts calling the handler",
+      description = "maximum number of (fractional) seconds to elapse after the first entry was queued before the queue starts calling the handler",
```
Comment (kikito) [inline kong/tools/queue_schema.lua:17]:
```suggestion
      description = "Maximum number of (fractional) seconds to elapse after the first entry was queued before the queue starts calling the handler.",
```

File: kong/tools/queue_schema.lua [hunk 44]
```diff
@@ -8,46 +8,46 @@ return Schema.define {
...
       -- This parameter has no effect if `max_batch_size` is 1, as queued entries will be sent
       -- immediately in that case.
     } },
     { max_entries = {
       type = "integer",
       default = 10000,
       between = { 1, 1000000 },
-      -- description = "maximum number of entries that can be waiting on the queue",
+      description = "maximum number of entries that can be waiting on the queue",
```
Comment (kikito) [inline kong/tools/queue_schema.lua:25]:
```suggestion
      description = "Maximum number of entries that can be waiting on the queue.",
```

File: kong/tools/queue_schema.lua [hunk 45]
```diff
@@ -8,46 +8,46 @@ return Schema.define {
...
     } },
     { max_bytes = {
       type = "integer",
       default = nil,
-      -- description = "maximum number of bytes that can be waiting on a queue, requires string content",
+      description = "maximum number of bytes that can be waiting on a queue, requires string content",
```
Comment (kikito) [inline kong/tools/queue_schema.lua:30]:
```suggestion
      description = "Maximum number of bytes that can be waiting on a queue, requires string content.",
```

File: kong/tools/queue_schema.lua [hunk 46]
```diff
@@ -8,46 +8,46 @@ return Schema.define {
...
     } },
     { max_retry_time = {
       type = "number",
       default = 60,
-      -- description = "time in seconds before the queue gives up calling a failed handler for a batch",
+      description = "time in seconds before the queue gives up calling a failed handler for a batch",
```
Comment (kikito) [inline kong/tools/queue_schema.lua:35]:
```suggestion
      description = "Time in seconds before the queue gives up calling a failed handler for a batch.",
```

File: kong/tools/queue_schema.lua [hunk 47]
```diff
@@ -8,46 +8,46 @@ return Schema.define {
...
       -- If this parameter is set to -1, no retries will be made for a failed batch
     } },
     {
       initial_retry_delay = {
         type = "number",
         default = 0.01,
         between = { 0.001, 1000000 }, -- effectively unlimited maximum
-        -- description = "time in seconds before the initial retry is made for a failing batch."
+        description = "time in seconds before the initial retry is made for a failing batch."
```
Comment (kikito) [inline kong/tools/queue_schema.lua:43]:
```suggestion
        description = "Time in seconds before the initial retry is made for a failing batch."
```

File: kong/tools/queue_schema.lua [hunk 48]
```diff
@@ -8,46 +8,46 @@ return Schema.define {
...
         -- For each subsequent retry, the previous retry time is doubled up to `max_retry_delay`
     } },
     { max_retry_delay = {
       type = "number",
       default = 60,
       between = { 0.001, 1000000 }, -- effectively unlimited maximum
-      -- description = "maximum time in seconds between retries, caps exponential backoff"
+      description = "maximum time in seconds between retries, caps exponential backoff"
```
Comment (kikito) [inline kong/tools/queue_schema.lua:50]:
```suggestion
      description = "Maximum time in seconds between retries, caps exponential backoff."
```

**Review by kikito — APPROVED** [review]
Approving with suggestions on queue fields

**Review by kikito — APPROVED** [review]
Approving with suggestions on queue fields

**Comment by outsinre** [comment]
This PR may introduce bug to EE, considering the EE schemas usually have different field attributes. Need careful cherry-pick.

File: kong/db/schema/metaschema.lua [hunk 49]
```diff
@@ -72,6 +72,7 @@ local field_schema = {
   { type = { type = "string", one_of = keys(Schema.valid_types), required = true }, },
   { required = { type = "boolean" }, },
   { reference = { type = "string" }, },
+  { description = { type = "string", len_min = 10, len_max = 500}, },
```
[the remark this answers is not in the document]
Comment (hbagdi) [reply kong/db/schema/metaschema.lua:75]: Temporary. See https://github.com/Kong/kong-ee/pull/5822

**Pull request merged**